### PR TITLE
RSDK-812-817: Extra params for Services

### PR DESCRIPTION
**Pull request by cheukt**

### CODE REVIEW

File: src/viam/services/sensors.py [hunk 1]
```diff
@@ -23,17 +23,21 @@ def __init__(self, name: str, channel: Channel):
         self.client = SensorsServiceStub(channel)
         self.name = name
 
-    async def get_sensors(self) -> List[ResourceName]:
+    async def get_sensors(self, extra: Optional[Mapping[str, Any]] = None) -> List[ResourceName]:
```
[the remark this answers is not in the document]
Comment (cheukt) [reply src/viam/services/sensors.py]: done

File: src/viam/services/vision.py [hunk 2]
```diff
@@ -134,52 +155,73 @@ async def get_detections(self, image: Union[Image, RawImage], detector_name: str
             confidence score of the labels, around the found objects in the next 2D image
             from the given camera, with the given detector applied to it.
         """
+        if extra is None:
+            extra = {}
         mime_type = CameraMimeType.JPEG
+        if isinstance(image, RawImage):
```
Comment (cheukt) [inline src/viam/services/vision.py:161]: @njooma these linting errors should've been found somehow with the earlier camera changes - is there anything we can do to make the linting more robust?

Comment (njooma) [reply src/viam/services/vision.py:161]: We can have much stricter mypy policies as part of our test suite, which should catch these statically typed issues

Comment (njooma) [reply src/viam/services/vision.py:161]: I'll add that to my todos

File: src/viam/services/vision.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import json
 from dataclasses import dataclass
 from enum import Enum
+from io import BytesIO
 from typing import Any, List, Mapping, Optional, Sequence, Union
 
 from grpclib.client import Channel
-from PIL.Image import Image
+from viam.media.viam_rgba_plugin import Image
```
Comment (njooma) [inline src/viam/services/vision.py:8]: This is just an alias for `from PIL import Image`. Not sure why you needed to import this (and then change all the `Image` definitions to `Image.Image`...

Comment (cheukt) [reply src/viam/services/vision.py:8]: @gvaradarajan says we need it to support the viam raw format (otherwise it isn't registered?)

Comment (njooma) [reply src/viam/services/vision.py:8]: Yuuup I see

**Review by njooma — APPROVED** [review]

File: src/viam/services/vision.py (hunk 2 above)
Comment (cheukt) [reply src/viam/services/vision.py:161]: thanks!

**Pull request merged**